### DISCUSSION TITLE
usb Babble

### DISCUSSION THREAD

**Discussion by yeliangchen2** [polarfire-soc/polarfire-soc-documentation#331] [Linux drivers for PolarFire Soc]
reboot system ,and the following error occurs occasionally.

**Comment by yeliangchen2** [comment]
![image](https://github.com/polarfire-soc/polarfire-soc-documentation/assets/16462894/ced9efb9-dce2-414c-8023-af9f0a7731c4)

**Reply by hughbreslin** [reply]
Do you have a usb connected from your PC to the USB otg connector? This can occur if USB is configured as host in Linux as your PC also acts as host. Host to host causes the babble print, in our default builds USB is configured as otg and in our Ubuntu release it’s configured as host. Which release are you using?

**Reply by yeliangchen2** [reply]
hello，thank you for your reply. USB is configured as host in Linux ,and the usb device is a 5G module. Restarting system too many times will occasionally bring up a babble.
By the way:  What is the function of the following. uncommenting this affects usb functionality.
![image](https://github.com/polarfire-soc/polarfire-soc-documentation/assets/16462894/d3c199d4-1c5b-4769-8e84-51a1d4193840)

**Reply by hughbreslin** [reply]
Hmm it may be a race condition and the USB isn’t configured correctly or there’s something outstanding before it comes up - are you doing soft restarts?

The USBInit is just configuring the DMA to support MSS addressing, have a look at the comment [here](https://github.com/polarfire-soc/hart-software-services/blob/master/init/hss_usb_init.c#L42)

**Reply by yeliangchen2** [reply]
yes, I am doing soft restart.  After doing soft restart too many times , it will occasionally bring up a babble.

**Reply by xiyanni** [reply]
Hi Yeliangchen2,

Could you try one way, such as delay actions after the UBS initation function?

**Reply by xiyanni** [reply]
Hi Yeliangchen2,

Could do two tests to check the results?
1. change the USB device, to check if this issue is caused by the 5G module.
2. Could you check you sequence in reboot, add delay or confirm the sequence may help. 
Please let me know what happens. Thanks.

**Reply by yeliangchen2** [reply]
the first one ,5G module. can not be hot plug.  the seconed ,for "check you sequence in reboot, add delay or confirm the sequence " ,what should I do. please give me specific steps, or examples

**Comment by xiyanni** [comment]
Hi Hugh,

Thanks for your supporting. If I understanding is correct, there are ways to keep USB configured correctly in restart. Could you share the ways to prevent the Babble issue?

**Reply by hughbreslin** [reply]
@xiyanni this isn't exactly what I meant - I've never seen this before in this context of multiple restarts, only in the context of a USB host controller (i.e PC) connected to an MPFS kit which is also configured as USB host. I'm not sure how to prevent this, my best suggestion is ensuring the USB is fully reset, it could be the case this isn't happening on a soft reset.
